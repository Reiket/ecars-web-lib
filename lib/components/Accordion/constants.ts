export const ACCORDION_TEST_ID = 'accordionTestId';
