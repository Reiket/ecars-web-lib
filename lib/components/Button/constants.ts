import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';