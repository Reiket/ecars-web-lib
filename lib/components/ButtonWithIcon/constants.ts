export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
