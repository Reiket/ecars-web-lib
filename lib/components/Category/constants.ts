export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
