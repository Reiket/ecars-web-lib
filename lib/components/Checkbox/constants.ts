import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
