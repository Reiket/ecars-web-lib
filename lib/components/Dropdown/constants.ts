import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
