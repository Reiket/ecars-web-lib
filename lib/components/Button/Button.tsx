import type {FC, MouseEvent, ReactNode} from 'react';
import {BUTTON_TEST_ID, ButtonAttributesType, ButtonColorType, ButtonSizeType} from '@/components/Button/constants';
import {cn} from '@/services/helpers';

export interface ButtonProps {
  size: ButtonSizeType;
  children: ReactNode;
  color: ButtonColorType;
  type?: ButtonAttributesType;
  disabled?: boolean;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  isTransparent?: boolean;
  withIcon?: boolean;
}

export const Button: FC<ButtonProps> = ({
  children,
  size,
  color,
  onClick,
  type = 'button',
  disabled = false,
  isTransparent = false,
  withIcon = false,
}) => {
  const classNames = cn('button', `button-${size}`, `button-${color}`, {
    'button-transparent': isTransparent,
    'button-icon': withIcon,
  });
  return (
    <button
      data-testid={BUTTON_TEST_ID}
      className={classNames}
      disabled={disabled}
      type={type}
      onClick={onClick}
    >
      {children}
    </button>
  );
};
