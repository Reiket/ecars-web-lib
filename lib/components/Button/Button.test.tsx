import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import {BUTTON_COLOR, BUTTON_SIZE, BUTTON_TEST_ID, ButtonAttributesType} from '@/components/Button/constants';
import {Button} from '@/components/Button/Button';
import '@testing-library/jest-dom';

describe('Button component', () => {
  afterEach(cleanup);

  const colors = Object.values(BUTTON_COLOR);
  const sizes = Object.values(BUTTON_SIZE);
  const types: ButtonAttributesType[] = ['submit', 'reset', 'button'];
  const props = {
    color: BUTTON_COLOR.GRAY,
    size: BUTTON_SIZE.BIG,
  };
  const disabledOptions = [true, false];
  colors.forEach((color) => {
    sizes.forEach((size) => {
      types.forEach((type) => {
        disabledOptions.forEach((disabled) => {
          const testName = `renders button with color ${color}, size ${size}, type ${type || 'button'} and disabled`;
          test(testName, () => {
            const {container} = render(
              <Button
                color={color}
                size={size}
                type={type}
                disabled={disabled}
              >
                button
              </Button>,
            );
            const buttonElement = screen.getByTestId(BUTTON_TEST_ID) as HTMLButtonElement;
            expect(buttonElement).toBeInTheDocument();
            expect(buttonElement).toHaveClass(`button--${color}`);
            expect(buttonElement).toHaveClass(`button--${size}`);
            expect(buttonElement.disabled).toBe(disabled);
            expect(buttonElement).toHaveStyle('cursor: default');
            expect(container).toMatchSnapshot();
          });
        });
      });
    });
  });
  test('changes button text according to the children text', () => {
    const {container, getByText, rerender} = render(<Button {...props}>button</Button>);
    let button = getByText('button');
    expect(button).toBeInTheDocument();

    rerender(<Button {...props}>Confirm</Button>);
    button = getByText('Confirm');
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('handles the click event', () => {
    const onClick = jest.fn();
    const {container, getByText} = render(
      <Button
        {...props}
        onClick={() => onClick()}
      >
        button
      </Button>,
    );
    fireEvent.click(getByText('button'));
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });

  test('the onClick event does not fire when button is disabled', () => {
    const onClick = jest.fn();
    const {container, getByText} = render(
      <Button
        {...props}
        disabled
        onClick={onClick}
      >
        button
      </Button>,
    );
    const button = getByText('button');
    fireEvent.click(button);
    expect(onClick).not.toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  test('additional classes are not applied if the button is disabled', () => {
    const {container, getByText} = render(
      <Button
        {...props}
        disabled
      >
        button
      </Button>,
    );
    const button = getByText('button');
    expect(button).toHaveClass('button');
    expect(container).toMatchSnapshot();
  });
  test('button has button-transparent if receive prop isTransparent', () => {
    const {container} = render(
      <Button
        {...props}
        isTransparent
      >
        button
      </Button>,
    );
    const button = screen.getByTestId(BUTTON_TEST_ID);
    expect(button).toHaveClass('button--transparent');
    expect(container).toMatchSnapshot();
  });
  test('button has button-icon if receive prop withIcon', () => {
    const {container} = render(
      <Button
        {...props}
        withIcon
      >
        button
      </Button>,
    );
    const button = screen.getByTestId(BUTTON_TEST_ID);
    expect(button).toHaveClass('button--icon');
    expect(container).toMatchSnapshot();
  });
});
