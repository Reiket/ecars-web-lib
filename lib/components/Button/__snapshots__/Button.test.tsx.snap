// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > additional classes are not applied if the button is disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > button has button-icon if receive prop withIcon 1`] = `
<div>
  <button
    class="button button--big button--gray button--icon"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > button has button-transparent if receive prop isTransparent 1`] = `
<div>
  <button
    class="button button--big button--gray button--transparent"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > changes button text according to the children text 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    Confirm
  </button>
</div>
`;

exports[`Button component > handles the click event 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color gray, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color green, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > renders button with color white, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component > the onClick event does not fire when button is disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component additional classes are not applied if the button is disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component button has button-icon if receive prop withIcon 1`] = `
<div>
  <button
    class="button button--big button--gray button--icon"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component button has button-transparent if receive prop isTransparent 1`] = `
<div>
  <button
    class="button button--big button--gray button--transparent"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component changes button text according to the children text 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    Confirm
  </button>
</div>
`;

exports[`Button component handles the click event 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color gray, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--gray"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color green, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--green"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type button and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type button and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type reset and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type reset and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type submit and disabled 1`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size big, type submit and disabled 2`] = `
<div>
  <button
    class="button button--big button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type button and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type button and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type reset and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type reset and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type submit and disabled 1`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size medium, type submit and disabled 2`] = `
<div>
  <button
    class="button button--medium button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type button and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type button and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type reset and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type reset and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="reset"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type submit and disabled 1`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    disabled=""
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component renders button with color white, size small, type submit and disabled 2`] = `
<div>
  <button
    class="button button--small button--white"
    data-testid="buttonTestId"
    type="submit"
  >
    button
  </button>
</div>
`;

exports[`Button component the onClick event does not fire when button is disabled 1`] = `
<div>
  <button
    class="button button--big button--gray"
    data-testid="buttonTestId"
    disabled=""
    type="button"
  >
    button
  </button>
</div>
`;
